server/src/test/integration/org/apache/cassandra/sidecar/testing/CassandraSidecarTestContext.java

```diff
@@ -197,6 +197,7 @@ public InstancesConfig refreshInstancesConfig()
     {
         // clean-up any open sessions or client resources
         close();
+        closeSessionProvider();
```

Comment:
`setInstancesConfig` create a new instance of session provider. Therefore, close the current one before creating a new one.